crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -332,4 +346,116 @@ mod dump_and_load_tests {
             panic!("Could not unpack the transaction from {loaded_transaction:?}");
         }
     }
+
+    #[tokio::test]
+    async fn dump_load_dump_load_on_exit() {
+        let dump_file = UniqueAutoDeletableFile::new("dump_load_dump_load_on_exit");
+        let devnet_dump_first = BackgroundDevnet::spawn_with_additional_args(&[
+            "--dump-path",
+            &dump_file.path,
+            "--dump-on",
+            "exit",
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        devnet_dump_first
+            .post_json("/create_block".into(), Body::from(json!({}).to_string()))
+            .await
+            .unwrap();
+        devnet_dump_first.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
```

Comment:
Why both block creation and minting?

Reply:
create a block is an event and minting is doing an invoke transaction, events like create a block and increase time have simple logic but transactions go through separate more complicated logic and I wanted to test both workflows